src/databricks/labs/dqx/geo/check_funcs.py

```diff
@@ -0,0 +1,468 @@
+
+    Note:
+        This function requires Databricks runtime 17.1 or above.
+    """
+    col_str_norm, col_expr_str, col_expr = _get_normalized_column_and_expr(column)
+    # NOTE: This function is currently only available in Databricks runtime 17.1 or above or in
+    #   Databricks SQL, due to the use of the `try_to_geometry` and `st_geometrytype` functions.
+    # TODO: Above mentioned functions are not (yet) available. Replace with equivalent functions
+    #   when available in OSS PySpark.
+    geom_cond = F.expr(f"try_to_geometry({col_str_norm}) IS NULL")
+    geom_type_cond = F.expr(f"st_geometrytype(try_to_geometry({col_str_norm})) <> '{MULTILINESTRING_TYPE}'")
+    condition = F.when(col_expr.isNull(), F.lit(None)).otherwise(geom_cond | geom_type_cond)
+    condition_str = f"` in column `{col_expr_str}` is not a multilinestring geometry"
+    return make_condition(
+        condition,
+        F.concat_ws("", F.lit("value `"), col_expr.cast("string"), F.lit(condition_str)),
+        f"{col_str_norm}_is_not_a_multilinestring",
+    )
+
+
+@register_rule("row")
+def is_multipolygon(column: str | Column) -> Column:
+    """Checks whether the values in the input column are multipolygon geometries.
+
+    Args:
+        column: column to check; can be a string column name or a column expression
+
+    Returns:
+        Column object indicating whether the values in the input column are multipolygon geometries
+
+    Note:
+        This function requires Databricks runtime 17.1 or above.
+    """
+    col_str_norm, col_expr_str, col_expr = _get_normalized_column_and_expr(column)
+    # NOTE: This function is currently only available in Databricks runtime 17.1 or above or in
+    #   Databricks SQL, due to the use of the `try_to_geometry` and `st_geometrytype` functions.
+    # TODO: Above mentioned functions are not (yet) available. Replace with equivalent functions
+    #   when available in OSS PySpark.
+    geom_cond = F.expr(f"try_to_geometry({col_str_norm}) IS NULL")
+    geom_type_cond = F.expr(f"st_geometrytype(try_to_geometry({col_str_norm})) <> '{MULTIPOLYGON_TYPE}'")
+    condition = F.when(col_expr.isNull(), F.lit(None)).otherwise(geom_cond | geom_type_cond)
+    condition_str = f"` in column `{col_expr_str}` is not a multipolygon geometry"
+    return make_condition(
+        condition,
+        F.concat_ws("", F.lit("value `"), col_expr.cast("string"), F.lit(condition_str)),
+        f"{col_str_norm}_is_not_a_multipolygon",
+    )
+
+
+@register_rule("row")
+def is_geometrycollection(column: str | Column) -> Column:
+    """Checks whether the values in the input column are geometrycollection geometries.
+
+    Args:
+        column: column to check; can be a string column name or a column expression
+
+    Returns:
+        Column object indicating whether the values in the input column are geometrycollection geometries
+
+    Note:
+        This function requires Databricks runtime 17.1 or above.
```

Comment:
This will not work in our github action env as we currently use 16.4 cluster. Maybe in serverless it will. Need to check. We may need to skip tests for functions that require 17.1